commons/src/store/threads.ts

```diff
@@ -44,6 +44,28 @@ function initializeThreads() {
       return threadsMap[queryKey];
     },
 
+    deleteThreads: async (
+      query: MailboxQuery,
+      threadsToDelete: Set<Thread>,
+    ) => {
+      const queryKey = JSON.stringify(query);
+      if (!threadsMap[queryKey] && (query.component_id || query.access_token)) {
+        threadsMap[queryKey] = (await fetchThreads(query)).map((thread) => {
+          thread.toString = () => thread.id;
+          return thread;
+        });
+      }
+      update((threads) => {
+        threadsToDelete.forEach((thread) => {
+          const deleteIndex = threadsMap[queryKey].indexOf(thread);
+          threadsMap[queryKey].splice(deleteIndex, deleteIndex + 1);
+        });
+        threads[queryKey] = threadsMap[queryKey];
+        return { ...threads };
+      });
+      return threadsMap[queryKey];
+    },
```

Comment:
I'm not sure if this is the most efficient way to do this, nor if it actually works because I don't know how to test without possibly deleting demo data (and I have no access to regenerate that data).

Reply:
Yeah, this is a good point. It should also be a restriction that we should apply on the middleware (e.g. if prop has "allow deletes" then only allow the endpoint to be called).

In the above code however, there is no new network request being made to delete a thread. Looking at https://developer.nylas.com/docs/api/#overview, we can't actually do this. 

At most, what we can do is changed the folder of the thread to the "trash" or "archive" folder.

@nylas/wonka we should have a discussion about this functionality.